lib/CacheAndBufferLayer.js

```diff
@@ -377,95 +375,84 @@ exports.Database.prototype.getSub = function (key, sub, callback) {
   });
 };
 
-exports.Database.prototype._callFlushDoneCallbacks = function (err) {
-  const callbacks = this._flushDoneCallbacks;
-  this._flushDoneCallbacks = []; // In case a callback calls flush().
-  for (const cb of callbacks) cb(err);
-};
-
 /**
  Writes all dirty values to the database
 */
-exports.Database.prototype.flush = function (callback = null) {
-  if (callback) this._flushDoneCallbacks.push(callback);
-  if (this.isFlushing) return;
-  const dirtyEntries = [];
-  for (const entry of this.buffer) {
-    if (entry[1].dirty) dirtyEntries.push(entry);
-  }
-  if (dirtyEntries.length === 0) {
-    setImmediate(() => this._callFlushDoneCallbacks());
-    return;
+exports.Database.prototype.flush = async function () {
+  if (this._flushDone == null) {
+    this._flushDone = (async () => {
+      for (;;) {
```

Comment:
Can we document what's going on with ``(;;)`` please?

Reply:
That's a pretty idiomatic way to do an infinite loop so I think a comment would do more harm than good. I generally prefer `while (true)` but I didn't feel like relaxing our lint rules to allow that because I'm lazy. :slightly_smiling_face:

Reply:
Tbf I have never seen `(;;)` before but I guessed at it's function.